### PR TITLE
[CPU] torch.gather for complex dtypes

### LINKED ISSUE #36340: torch.gather doesn't work for complex tensors
>>> torch.gather(t, 1, torch.tensor([[0,0],[1,0]]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: method_name not implemented for 'ComplexFloat'

cc @ezyang @anjali411 @dylanbespalko

### PR DESCRIPTION
**Pull request by nuka137**
This PR resolves #36340 .

### CODE REVIEW

**Comment by nuka137** [comment]
@anjali411 

I created the issue which resolves #36340 .

The error will not be output by this PR.
But I'm not sure whether we need to implement test as well.

**Comment by nuka137** [comment]
@anjali411 

Sorry for late response.
I implemented complex64 and complex128 tests as well.
Could you check them?

File: test/test_torch.py [hunk 1]
```diff
@@ -2593,6 +2593,16 @@ def _test_gather(self, cast, test_bounds=True):
         t = torch.tensor([[False, True], [True, True]])
         self.assertEqual(torch.gather(t, 1, torch.tensor([[0, 0], [1, 0]])), torch.tensor([[False, False], [True, True]]))
 
+        # Complex64, Complex128 test case
```
Comment (anjali411) [inline test/test_torch.py:2596]: I think we should just do the same thing as for float above. prob loop it for complex and float types

Comment (anjali411) [reply test/test_torch.py:2596]: https://github.com/pytorch/pytorch/blob/3b832ee2bfea70ecee65261419178165d02e3131/test/test_torch.py#L2540-L2557

Comment (nuka137) [reply test/test_torch.py]: @anjali411 

Thanks for reviewing.
I fixed the complex64 and complex128 tests to do the same thing of float.

**Comment by nuka137** [comment]
@anjali411 

Is there any updates about this PR?
I have already fixed the issue you mentioned.

**Comment by anjali411** [comment]
Hi @nuka137 thanks for the follow up! There are some test failures because the gather test is being run on cuda for complex dtypes (which is not supported yet) so you can add something like:

```
if device.startswith('cuda') and dtype.is_complex:
                self.assertRaises(RuntimeError, lambda: torch.gather(...))
```

**Comment by nuka137** [comment]
@anjali411 

Thanks!
I rebased to the latest master branch and pushed to my branch.
One test is still failed but I think it is not related to this PR.
Could you check it?

**Review by anjali411 — APPROVED** [review]
looks like gather was added for complex on cuda and the PR just needed to be rebased. thanks for the PR! :D

**Pull request closed without merging**